fitlins/interfaces/bids.py

```diff
@@ -361,6 +361,14 @@ def _run_interface(self, runtime):
         mask_files = []
         entities = []
         for ents in self.inputs.entities:
+            if 'run' in ents:
+                # Case: run is an integer (01, 1, etc.)
+                if isinstance(ents['run'], (int, float)):
+                    ents['run'] = [ents['run'], None]
+                # Case: run is a list with one element [1]
+                elif isinstance(ents['run'], list) and len(ents['run']) == 1:
+                    ents['run'].append(None)
```

Comment:
I think we can simplify, and we should leave a comment explaining why this is being done.

```suggestion
            # Hack around implicit "run-1" being added to files without run entities
            # This problem may be worth addressing in spec generation instead, but doing it here for now
            # See https://github.com/poldracklab/fitlins/pull/411
            if ents.get('run') in (1, [1]):
                ents['run'] = [1, None]
```

Could you check whether this patch works for your use case?

Reply:
Thanks. Will test on two separate datasets that do and don't include the edge case.

Reply:
@effigies - this worked in both test cases ds000009 and ds003858